agent/consul/gateways/bind.go

```diff
@@ -71,3 +84,52 @@ func getReferences(route structs.BoundRoute) (referenceSet, gatewayRefs) {
 
 	return parentRefs, gatewayRefs
 }
+
+func requestToResourceRef(req controller.Request) structs.ResourceReference {
+	ref := structs.ResourceReference{
+		Kind: req.Kind,
+		Name: req.Name,
+	}
+	if req.Meta != nil {
+		ref.EnterpriseMeta = *req.Meta
+	}
+	return ref
+}
+
+// RemoveGateway sets the route status for the given gateway to be unbound if it should be bound
+func RemoveGateway(gateway structs.ResourceReference, entries ...structs.BoundRoute) []structs.ControlledConfigEntry {
```

Comment:
This is triggered when we're cleaning up resources after a gateway has been deleted. We need to be able to set any route that references a gateway to now say, "I'm not bound to this gateway anymore since it's been deleted".